### PR TITLE
items: can request api for an item

### PR DESCRIPTION
**Pull request by BadrAly**
* Removes duplicate fixtures function.

Co-Authored-by: Aly Badr <aly.badr@rero.ch>

## Why are you opening this PR?

- Which task/US does it implement?
https://tree.taiga.io/project/rero21-reroils/us/657?milestone=254793

## How to test?

- To check if an item can be requested for a given patron barcode and a transaction library PID

`~/item/<item_pid>/can_request/<library_pid>/<patron_barcode>`
API response in this format:
`can_request`: True or False
`reason`:  the reason why the item can not be requested

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

File: rero_ils/modules/items/api_views.py [hunk 1]
```diff
@@ -316,3 +316,53 @@ def item_availability(item_pid):
...
+    if not library:
+        return jsonfiy_response(False, 'Library not found.')
+    # Create a loan
+    loan = {
+        'patron_pid': patron.pid, 'item_pid': item.pid,
+        'library_pid': library_pid}
+    loan = Loan(loan)
+    if not can_be_requested(loan):
+        return jsonfiy_response(
+            False, 'Circ policies do not allow request on this item.')
```
Comment (Garfield-fr) [inline rero_ils/modules/items/api_views.py]: `Circulation policy do not allow request on this item.`

File: tests/fixtures/circulation.py [hunk 2]
```diff
@@ -406,24 +406,6 @@ def system_librarian_sion_no_email(
...
-    """Create Martigny patron without sending reset password instruction."""
-    ptrn = Patron.create(
-        data=patron_martigny_data,
-        delete_pid=False,
-        dbcommit=True,
-        reindex=True)
-    flush_index(PatronsSearch.Meta.index)
-    return ptrn
-
-
```
Comment (zannkukai) [inline tests/fixtures/circulation.py]: Why removing this fixture ?

Comment (BadrAly) [reply tests/fixtures/circulation.py]: It is created two times in the same file. I removed the duplicate.

File: rero_ils/modules/items/utils.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+from flask_babelex import gettext as _
+
+from .api import Item, ItemStatus
+from ..libraries.api import Library
+from ..loans.api import Loan
+from ..loans.utils import can_be_requested
+from ..patrons.api import Patron
+
+
+def jsonfiy_response(response=False, reason=None):
```
Comment (AoNoOokami) [inline rero_ils/modules/items/utils.py]: jsonfiy instead of jsonify

File: rero_ils/modules/items/utils.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    return jsonify({
+        'can_request': response,
+        'reason': reason
+    })
+
+
+def can_request_item(item_pid, library_pid, patron_barcode):
+    """Check if a librarian can request an item for a patron.
+
+    required_parameters: item_pid, library_pid, patron_barcode
```
Comment (zannkukai) [inline rero_ils/modules/items/utils.py]: Method name is not relevant ; could be as `is_librarian_can_request_item` or `can_request_as_patron`

Comment (BadrAly) [reply rero_ils/modules/items/utils.py]: renamed to `is_librarian_can_request_item_for_patron`

**Pull request merged**